### PR TITLE
Lab3 letter frequency

### PR DESCRIPTION
**Pull request by vinilapeddireddy**
Fixes #63 

#### Changes proposed in this PR:
* Implemented letter frequency
* Write a function that returns a sorted slice of strings with elements "{key}:{val}"
* Add unit test

### CODE REVIEW

File: 03_letters/vinilapeddireddy/main.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	"sort"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprint(out, strings.Join(sortLetters(letters("aba")), "\n"))
+}
+func sortLetters(m map[rune]int) []string {
```
Comment (juliaogris) [inline 03_letters/vinilapeddireddy/main.go:17]: Blank line between functions.

File: 03_letters/vinilapeddireddy/main.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	keys := []int{}
+	for k := range m {
+		keys = append(keys, int(k))
+	}
+	sort.Ints(keys)
+	sortedKeys := make([]string, 0, len(keys))
+	for _, k := range keys {
+		sortedKeys = append(sortedKeys, fmt.Sprintf("%s : %d", string(k), m[rune(k)]))
```
Comment (juliaogris) [inline 03_letters/vinilapeddireddy/main.go]: `"%s:%d"`

File: 03_letters/vinilapeddireddy/main_test.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+	// Then
+	expected := `a : 2
+b : 1`
+	actual := buf.String()
+	r.Equalf(expected, actual, "Unexpected output in main()")
+}
+
+func TestLetters(t *testing.T) {
+
```
Comment (juliaogris) [inline 03_letters/vinilapeddireddy/main_test.go]: Remove blank line.

**Review by juliaogris — CHANGES_REQUESTED** [review]
Very close, looking good!

Rebase onto master.
Update commits not to include any mention of fibonacci and use imperative mood (see [slides](https://gotraining.xyz/gotraining.slide#29) for rebasing help.)

File: 03_letters/vinilapeddireddy/main.go (hunk 1 above)
Comment (vinilapeddireddy) [reply 03_letters/vinilapeddireddy/main.go:17]: Done

File: 03_letters/vinilapeddireddy/main.go (hunk 2 above)
Comment (vinilapeddireddy) [reply 03_letters/vinilapeddireddy/main.go]: Done

File: 03_letters/vinilapeddireddy/main_test.go (hunk 3 above)
Comment (vinilapeddireddy) [reply 03_letters/vinilapeddireddy/main_test.go]: Done

**Review by juliaogris — APPROVED** [review]
LGTM 😻

**Comment by juliaogris** [comment]
Once you have approval you should be able to just click the `Update branch` button and without further reapproval merge.

**Pull request merged**